File: maximum-number-of-balloons/maximum-number-of-balloons.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        hash_map = Counter(text)
        ans = 0
        while (hash_map['b'] >= 1 and hash_map['a'] >= 1 and hash_map['l'] >= 2 and hash_map['o'] >= 2 and hash_map['n'] >= 1):
            ans += 1
            hash_map['b'] -= 1
            hash_map['a'] -= 1
            hash_map['l'] -= 2
            hash_map['o'] -= 2
            hash_map['n'] -= 1
        
        return ans

File: maximum-number-of-balloons/maximum-number-of-balloons.cpp
class Solution {
public:
    int maxNumberOfBalloons(string text) {
        unordered_map<char, int> counter;
        for (const char& c: text) ++counter[c];
        int ans = 0;
        while(counter['b'] >= 1 && counter['a'] >= 1 && counter['l'] >= 2 && counter['o'] >= 2 \
              && counter['n'] >= 1) {
            ++ans; --counter['b']; --counter['a'];
            counter['l'] -= 2;
            counter['o'] -= 2;
            --counter['n'];
        }
        return ans;
    }
};